src/worker.ts

```diff
@@ -87,11 +100,43 @@ const registerWebhook = async (event: Event, requestUrl: URL, suffix: string, se
  * Remove webhook
  * https://core.telegram.org/bots/api#setwebhook
  */
-const unRegisterWebhook = async (event: Event) => {
+const unRegisterWebhook = async () => {
   const r = await (await fetch(apiUrl("setWebhook", { url: "" }))).json();
   return new Response("ok" in r && r.ok ? "Ok" : JSON.stringify(r, null, 2));
 };
 
+const onBotInstall = async (event: MyChatQueryType) => {
+  const status = event.new_chat_member.status;
+  const triggerUserName = event.new_chat_member.user.username;
+  const chatId = event.chat.id;
+  const fromId = event.from.id;
+  const groupName = event.chat.title;
+
+  const botName = await getBotUsername();
+
+  console.log(status, chatId, fromId, groupName);
+
+  if (botName === triggerUserName) {
+    // true if this is a valid bot install and uninstall
+    switch (status) {
+      case "kicked":
+        await isBotRemoved(chatId, fromId);
+        break;
+      case "left":
+        await isBotRemoved(chatId, fromId);
+        break;
```

Comment:
Does seem to be working when the bot is removed. Probably unnecessary?

Reply:
> Yea, we would do something on bot remove in the future.. i had the bot delete data on remove but @pavlovcik did not want it deleted.. But that should come in handy pretty soon

What is the "data" in this context? I don't recall. 

But yes generally speaking I don't think it's a good idea to ever delete data. "Soft deletes" where they are hidden from the user for example are okay.

Reply:
> What is the "data" in this context? I don't recall.

Group and repo mapped to the group was originally deleted but we need these data not deleted.. the event is still open, as we are adding more features we'd soft delete some things